dace/codegen/targets/cpu.py

```diff
@@ -1935,19 +1935,33 @@ def _generate_Reduce(self, sdfg, dfg, state_id, node, function_stream,
         )
 
         # If axes were not defined, use all input dimensions
-        input_dims = input_memlet.subset.dims()
-        output_dims = output_memlet.subset.data_dims()
+        input_num_dims = input_memlet.subset.dims()
         if axes is None:
-            axes = tuple(range(input_dims))
+            axes = tuple(range(input_num_dims))
+        output_num_dims = input_num_dims - len(axes)
 
         # Obtain variable names per output and reduction axis
         axis_vars = []
+        output_axis_vars = dict()
+        output_dims = []
         octr = 0
-        for d in range(input_dims):
+        input_subset = input_memlet.subset
+        output_subset = output_memlet.subset
+        for d in range(input_num_dims):
             if d in axes:
                 axis_vars.append("__i%d" % d)
             else:
                 axis_vars.append("__o%d" % octr)
+                ri = input_subset[d]
+                rngi = ri[1] - ri[0] + 1
```

Comment:
use subset.size()

Reply:
Implemented in  47f8c7046c1d6955fa21e2ada2ad8bc19cd975e1